### PR TITLE
 Feature: Added instrument paramater snapshot_exclude 

### PR DESCRIPTION
**Pull request by nyxus**
Changes proposed in this pull request:
- Added a parameter to the instrument parameter to exclude it from the snapshot. 

Note that this is very similar to #1651, the goal is the same but it has a different implementation. To see the reason why we need this please look at #1651 which has a detailed problem description. This implementation was suggested by @jenshnielsen, which uses the instrument parameter to store the `snapshot_exclude` instead of providing a list to the snapshot function. 

@jenshnielsen, @AdriaanRol

### CODE REVIEW

**Comment by nyxus** [comment]
Note that I still need to add tests.

**Comment by nyxus** [comment]
I've added tests and also tested the new implementation on a device on my desk. Can someone review the code? @jenshnielsen, @WilliamHPNielsen?

File: qcodes/instrument/base.py [hunk 1]
```diff
@@ -190,19 +190,22 @@ def snapshot_base(self, update: bool=False,
...
+                snap['parameters'][name] = param.snapshot(update=update_par)
             except:
                 # really log this twice. Once verbose for the UI and once
                 # at lower level with more info for file based loggers
-                self.log.warning(f"Snapshot: Could not update parameter: "
-                                 f"{name}")
-                self.log.info(f"Details for Snapshot:",
-                              exc_info=True)
-
+                logging.info("Snapshot: Could not update parameter: {}".format(name))
```
Comment (jenshnielsen) [inline qcodes/instrument/base.py]: Please change this back to `self.log.warning`

File: qcodes/tests/test_snapshot.py [hunk 2]
```diff
@@ -37,6 +37,44 @@ def test_snapshot_skip_params_update(request, params, params_to_skip):
...
+    for param_exl in params_to_exclude:
+        inst.parameters[param_exl].snapshot_exclude = True
+
+    snap = inst.snapshot()
+    snap_params = [x for x in snap["parameters"]]
+    params = [x for x in params if x not in params_to_exclude]
+
+    assert all(elem not in snap_params for elem in params_to_exclude), f"Parameter(s) is not excluded from the " \
+        f"snapshot. expected: {params}, actual: {snap_params}"
+
```
Comment (jenshnielsen) [inline qcodes/tests/test_snapshot.py]: Would it make sense to extend this to also test that all of the parameters in `params` are indeed in the snapshot?

File: qcodes/instrument/base.py (hunk 1 above)
Comment (nyxus) [reply qcodes/instrument/base.py]: Done

File: qcodes/tests/test_snapshot.py (hunk 2 above)
Comment (nyxus) [reply qcodes/tests/test_snapshot.py]: I was thinking the same thing. And the code is so simple nor hurts it, so added to the test.

**Comment by nyxus** [comment]
I've adapted the code after review and fixed the line length issues.

File: qcodes/instrument/parameter.py [hunk 3]
```diff
@@ -156,6 +156,12 @@ class _BaseParameter(Metadatable):
         snapshot_value (Optional[bool]): False prevents parameter value to be
             stored in the snapshot. Useful if the value is large.
 
+        snapshot_exclude (Optional[bool]): True prevents parameter to be
+            included in the snapshot. Useful if there are many of the same
+            parameter which are clogging up the snapshot.
+
```
Comment (jenshnielsen) [inline qcodes/instrument/parameter.py]: Please remove this blank line. I think this will render incorrectly in the html of the docs otherwise

**Comment by jenshnielsen** [comment]
Looks good. I will leave this for a bit to givee others a chance to merge

File: qcodes/instrument/parameter.py (hunk 3 above)
Comment (nyxus) [reply qcodes/instrument/parameter.py]: Done!

**Pull request merged**